test, build: Separate `read_json` function into its own module

hebasto opened the pull request:
Currently, 4 source files rely on the definition of the `read_json` function provided in `src/test/script_tests.cpp`.

This PR breaks this entanglement, improves code structure and maintainability.

MarcoFalke commented on src/test/util/json.cpp:
```diff
@@ -0,0 +1,20 @@
+{
+    UniValue v;
+    if (!v.read(jsondata) || !v.isArray()) {
+        BOOST_ERROR("Parse error.");
```
In theory it is not allowed to link boosttest to libtest_util, but given that CI didn't fail, `BOOST_ERROR` is probably header-only?

hebasto replied:
39e66e938fb688f5400ad94a1b317fcc2a87bc31 from #24301 ?

MarcoFalke replied:
Ah ok. I think I have a slight preference to keep boost out of libtest_util, and thus out of the fuzz tests

hebasto replied:
What about just moving the `read_json` function into the `src/test/main.cpp` source file?

hebasto replied:
> Ah ok. I think I have a slight preference to keep boost out of libtest_util, and thus out of the fuzz tests

We already have: https://github.com/bitcoin/bitcoin/blob/ea67232cdb80c4bc3f16fcd823f6f811fd8903e1/src/test/util/chainstate.h#L17

fanquake replied:
We can probably remove that, and keep it Boost free.

MarcoFalke replied:
> We already have: 

This can probably be replaced by a LogPrint or so

fanquake replied:
Done in #26009.

fanquake replied:
Now that #26009 has been merged, this should be reworked to avoid Boost usage.

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/25974#issuecomment-1253909158).

MarcoFalke commented on src/test/util/json.cpp:
```diff
@@ -0,0 +1,20 @@
+{
+    UniValue v;
+    if (!v.read(jsondata) || !v.isArray()) {
+        BOOST_ERROR("Parse error.");
```
Might be better to use `Assert` in any case to avoid the dead, confusing and dangerous return? (Dangerous because it returns an empty array, potentially running no tests at all)

hebasto commented:
Updated 57c9bfabe6a1cfda2a179a457ab93d03b279ee30 -> aa326abcfc63d842e77feb5e3cffc19926369416 ([pr25974.01](https://github.com/hebasto/bitcoin/commits/pr25974.01) -> [pr25974.02](https://github.com/hebasto/bitcoin/commits/pr25974.02)):

- addressed reviewers' comments

hebasto commented:
Rebased aa326abcfc63d842e77feb5e3cffc19926369416 -> 3394ad21b5a7e8ddfcc817e2eee996fb4b0bd4c6 ([pr25974.02](https://github.com/hebasto/bitcoin/commits/pr25974.02) -> [pr25974.03](https://github.com/hebasto/bitcoin/commits/pr25974.03)) due to the conflict with #26286.

MarcoFalke commented on src/test/util/json.cpp:
```diff
@@ -0,0 +1,18 @@
+
+#include <test/util/json.h>
+
+#include <boost/test/unit_test.hpp>
```
why?

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/25974#issuecomment-1406242932).

hebasto committed:
test, build: Separate `read_json` function into its own module

hebasto commented:
Rebased 3394ad21b5a7e8ddfcc817e2eee996fb4b0bd4c6 -> 7a820cee0e6408f5848799011d82dd29ee7fa8c5 ([pr25974.03](https://github.com/hebasto/bitcoin/commits/pr25974.03) -> [pr25974.04](https://github.com/hebasto/bitcoin/commits/pr25974.04)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/25974#discussion_r1084031360)

fanquake reviewed (approved):
ACK 7a820cee0e6408f5848799011d82dd29ee7fa8c5
